core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java

```diff
@@ -73,6 +76,9 @@ public DatasourceService( IMetadataDomainRepository metadataDomainRepository,
     this.mondrianCatalogService = mondrianCatalogService;
     this.repositoryFileAclAdapter = repositoryFileAclAdapter;
     this.pluginResourceLoader = pluginResourceLoader;
+    if ( metadataDomainRepository instanceof IDataSourceAwareMetadataDomainRepository ) {
```

Comment:
checking if metadataDomainRepository supports new interface IDataSourceAwareMetadataDomainRepository here in parent class instead of sub classes MetadataService/DataSourceWizardService. Could have done separately in MetadataService/DataSourceWizardService but chose this as the better option.